WooCommerce/Classes/ViewRelated/Products/Edit Product/Product Settings/Catalog Visibility/ProductCatalogVisibilityViewController.swift

```diff
@@ -0,0 +1,234 @@
+            return nil
+        }
+
+        headerView.leftText = nil
+        headerView.rightText = nil
+
+        return headerView
+    }
+}
+
+// MARK: - Support for UITableViewDataSource
+//
+private extension ProductCatalogVisibilityViewController {
+    /// Configure cellForRowAtIndexPath:
+    ///
+   func configure(_ cell: UITableViewCell, for row: Row, at indexPath: IndexPath) {
+        switch cell {
+        case let cell as SwitchTableViewCell:
+            configureFeaturedProduct(cell: cell)
+        case let cell as BasicTableViewCell:
+            configureCatalogVisibilitySelector(cell: cell, indexPath: indexPath)
+        default:
+            fatalError("Unidentified product catalog visibility row type")
+        }
+    }
+
+    func configureFeaturedProduct(cell: SwitchTableViewCell) {
+        cell.title = NSLocalizedString("Featured Product", comment: "Featured Product switch in Product Catalog Visibility")
+        cell.isOn = productSettings.featured
+        cell.onChange = { [weak self] value in
+            self?.productSettings.featured = value
+        }
+    }
+
+    func configureCatalogVisibilitySelector(cell: BasicTableViewCell, indexPath: IndexPath) {
+        let row = sections[indexPath.section].rows[indexPath.row]
+        cell.selectionStyle = .default
+        cell.textLabel?.text = row.catalogVisibility?.description
+        cell.accessoryType = productSettings.catalogVisibility == row.catalogVisibility ? .checkmark : .none
+    }
+}
+
+// MARK: - Constants
+//
+extension ProductCatalogVisibilityViewController {
+
+    /// Table Rows
+    ///
+    enum Row {
+        /// Listed in the order they appear on screen
+        case featuredProduct
+        case visible
+        case catalog
+        case search
+        case hidden
+
+        var reuseIdentifier: String {
+            switch self {
+            case .featuredProduct:
+                return SwitchTableViewCell.reuseIdentifier
+            case .visible, .catalog, .search, .hidden:
+                return BasicTableViewCell.reuseIdentifier
+            }
+        }
+
+        var catalogVisibility: ProductCatalogVisibility? {
+            switch self {
+            case .visible:
+                return .visible
+            case .catalog:
+                return .catalog
+            case .search:
+                return .search
+            case .hidden:
+                return .hidden
+            default:
+                return nil
+            }
+        }
+    }
+
+    /// Table Sections
+    ///
+    struct Section {
+        let title: String?
+        let footer: String?
+        let rows: [Row]
+
+        init(title: String? = nil, footer: String? = nil, rows: [Row]) {
+            self.title = title
+            self.footer = footer
+            self.rows = rows
+        }
+
+        init(title: String? = nil, rightTitle: String? = nil, footer: String? = nil, row: Row) {
```

Comment:
the `footer` in these 2 initializers and `rightTitle` in this initializer don't seem to be used, do you think we can remove them?

Reply:
I removed `title` and `rightTitle`, since `footer` is now used.